### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by stillsheryl**
This PR will:
- Improve test coverage with more edge case testing
- Pre-fill form with current information when editing shelter or pet
- Add lists to pets
- All tests passing

### CODE REVIEW

File: app/controllers/pets_controller.rb [hunk 1]
```diff
@@ -13,7 +13,8 @@ def create
       age: params[:age],
       sex: params[:sex],
       image: params[:image],
-      shelter_id: params[:id]
+      shelter_id: params[:id],
+      status: params[:status]
```
Comment (stillsheryl) [inline app/controllers/pets_controller.rb:17]: nice addition here

File: app/views/pets/edit.html.erb [hunk 2]
```diff
@@ -17,15 +17,17 @@
   <h1>Update Pet Information</h1>
   <%= form_tag "/pets/#{params[:id]}", method: :patch do %>
       <%= label_tag :name %>
-      <%= text_field_tag :name %><br/><br/>
+      <%= text_field_tag :name, @pet.name %><br/><br/>
```
Comment (stillsheryl) [inline app/views/pets/edit.html.erb:20]: nice idea to pre-fill the forms

File: spec/features/pets/index_spec.rb [hunk 3]
```diff
@@ -19,28 +19,28 @@
...
     expect(page).to have_content(pet_2.age)
     expect(page).to have_content(pet_2.sex)
+    expect(page).to have_css("img[src*='https://images.unsplash.com/photo-1548681528-6a5c45b66b42?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=634&q=80']")
+    expect(page).to have_content(pet_2.shelter.name)
     expect(page).to have_content(pet_3.name)
     expect(page).to have_content(pet_3.age)
     expect(page).to have_content(pet_3.sex)
-    expect(page).to have_content(pet_1.shelter.name)
-    expect(page).to have_content(pet_2.shelter.name)
+    expect(page).to have_css("img[src*='https://images.unsplash.com/photo-1513360371669-4adf3dd7dff8?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1050&q=80']")
```
Comment (stillsheryl) [inline spec/features/pets/index_spec.rb:42]: nice to have different links to test. good catch!

**Review by stillsheryl — COMMENTED** [review]
Looks good!
